File: solutions/0003-longest-substring-without-repeating-characters/solution.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        max_len = 0
        i = 0
        j = 1
        if len(s) == 0:
            return 0
        if len(s) == 1:
            return 1
        dup = [s[i]]
        while i<len(s) and j < len(s):
            if s[j] in dup:
                dup.remove(s[i])
                i+=1
            else:
                dup.append(s[j])
                j+=1
            if max_len < len(dup):
                max_len = len(dup)
        return max_len
                
            


        

File: solutions/0003-longest-substring-without-repeating-characters/solution.java
import java.util.*;

class Solution {
    public int lengthOfLongestSubstring(String s) {
        int n = s.length();
        int i = 0;
        int ans = 0;

        Set<Character> set = new HashSet<>();

        for(int j = 0; j < n; j++){
            
            while(set.contains(s.charAt(j))){
                set.remove(s.charAt(i));
                i++;
            }

            set.add(s.charAt(j));

            ans = Math.max(ans, set.size());
        }

        return ans;
    }
}
